pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -161,51 +169,62 @@ func (p *Plugin) getConfig() (*pluginConfig, error) {
 	return p.config, nil
 }
 
-func (p *Plugin) setConfig(config *pluginConfig) error {
+func (p *Plugin) setConfig(config *pluginConfig) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.config = config
+}
+
+func (p *Plugin) getClients() ([]kubeClient, error) {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	if p.clients == nil {
+		return nil, status.Error(codes.FailedPrecondition, "not configured")
+	}
+	return p.clients, nil
+}
+
+func (p *Plugin) setClients(clients []kubeClient) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	// Start watcher to set CA Bundle in objects created after server has started
-	var cancelWatcher func()
+	p.clients = clients
+}
+
+// startInformers creates informers to set CA Bundle in objects created after server has started
+func (p *Plugin) startInformers(config *pluginConfig, clients []kubeClient) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	stopCh := make(chan struct{})
 	if config.WebhookLabel != "" || config.APIServiceLabel != "" {
-		ctx, cancel := context.WithCancel(context.Background())
-		watcher, err := newBundleWatcher(ctx, p, config)
-		if err != nil {
-			cancel()
-			return err
-		}
-		var wg sync.WaitGroup
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			if err := watcher.Watch(ctx); err != nil && !errors.Is(err, context.Canceled) {
-				p.log.Error("Unable to watch", "error", err)
+		informerSynced := []cache.InformerSynced{}
+		for _, client := range clients {
+			informer := client.Informer(p.informerEvent)
+			if informer != nil {
+				go informer.Run(stopCh)
+				informerSynced = append(informerSynced, informer.HasSynced)
 			}
-		}()
-		cancelWatcher = func() {
-			cancel()
-			wg.Wait()
 		}
+		cache.WaitForCacheSync(stopCh, informerSynced...)
```

Comment:
Also, if/when this becomes cancel-able, we should probably check the return value and bail if it fails (i.e. the context was  canceled)

Reply:
added this check in this commit: https://github.com/spiffe/spire/pull/2857/commits/96f766aa6dff2b94462b69f910d7b753cf35e828